adapter/internal/synchronizer/apis_fetcher.go

```diff
@@ -43,6 +43,14 @@ const (
 	defaultCertPath string = "/home/wso2/security/controlplane.pem"
 )
 
+func init() {
+	conf, _ := config.ReadConfigs()
+	sync.InitializeWorkerPool(conf.ControlPlane.RequestWorkerPool.PoolSize, conf.ControlPlane.RequestWorkerPool.QueueSizePerPool,
+		conf.ControlPlane.RequestWorkerPool.PauseTimeAfterFailure, conf.Adapter.Truststore.Location,
+		conf.ControlPlane.SkipSSLVerification, conf.ControlPlane.HTTPClient.RequestTimeOut, conf.ControlPlane.RetryInterval,
+		conf.ControlPlane.ServiceURL, conf.ControlPlane.Username, conf.ControlPlane.Password)
+}
```

Comment:
Why don't we init this within the package synchronizer. So we do not want to pass config params, inside the func `sync.InitializeWorkerPool` we can load `conf, _ := config.ReadConfigs()` and read required params there?

Reply:
that is because it is a common package. (pkg) In other words it is reused in some other repos. If include config read inside, we won't be able to do that.